python/ray/ml/train/integrations/huggingface/huggingface_trainer.py

```diff
@@ -0,0 +1,189 @@
+from typing import Any, Callable, Optional, Dict, Union, Type
+
+from transformers.trainer import Trainer
+from torch.utils.data import Dataset
+
+from ray.train.torch import TorchConfig
+from ray.train.utils import construct_train_func
+from ray.ml.trainer import GenDataset
+from ray.ml.train.integrations.torch import TorchTrainer
+from ray.ml.config import ScalingConfig, RunConfig
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+from ray.util import PublicAPI
+
+
+def _huggingface_train_loop_per_worker(config: Dict[str, Any]):
+    config
+    return
+
+
+@PublicAPI(stability="alpha")
+class HuggingFaceTrainer(TorchTrainer):
+    """A Trainer for data parallel HuggingFace Transformers on PyTorch training.
+
+    This Trainer runs the ``transformers.Trainer.train()`` method on multiple
+    Ray Actors. The training is carried out in a distributed fashion through PyTorch
+    DDP. These actors already have the necessary torch process group already
+    configured for distributed PyTorch training.
+
+    The training function ran on every Actor will first run the
+    specified ``trainer_init_per_worker`` function to obtain an instantiated
+    ``transformers.Trainer`` object. The ``trainer_init_per_worker`` function
+    will have access to preprocessed train and evaluation datsets.
+
+    If the ``datasets`` dict contains a training dataset (denoted by
+    the "train" key), then it will be split into multiple dataset
+    shards, with each Actor training on a single shard.
+    All the other datasets will not be split.
+
+    Please note that if you use a custom ``transformers.Trainer`` subclass,
+    the ``get_train_dataloader`` method will be overriden to disable distributed
+    sampling, as the dataset will already be sharded.
+
+    Hugging Face loggers will be automatically disabled, and the ``local_rank``
+    argument in ``TrainingArguments`` will be automatically set.
+
+    Example:
+        .. code-block:: python
+            # Based on
+            # huggingface/notebooks/examples/language_modeling_from_scratch.ipynb
+
+            # Hugging Face imports
+            from datasets import load_dataset
+            from transformers import AutoConfig, AutoModelForCausalLM, AutoTokenizer
+            from transformers import Trainer, TrainingArguments
```

Comment:
combine the 2? everything is from transformers package.
actually a random minor recommendation is that given how many Trainers, and Configs there are below, it may be clearer to just use transoformers.Trainer, transformers.AutoConfig below.